src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -125,9 +120,44 @@ create_distributed_table(PG_FUNCTION_ARGS)
 	text *distributionColumnText = PG_GETARG_TEXT_P(1);
 	Oid distributionMethodOid = PG_GETARG_OID(2);
 
+	text *colocateWithTableNameText = PG_GETARG_TEXT_P(3);
+	char *colocateWithTableName = text_to_cstring(colocateWithTableNameText);
+
 	char *distributionColumnName = text_to_cstring(distributionColumnText);
 	char distributionMethod = LookupDistributionMethod(distributionMethodOid);
 
+	/* check if we try to colocate with hash distributed tables */
+	if (strncmp(colocateWithTableName, "default", NAMEDATALEN) != 0 &&
+		strncmp(colocateWithTableName, "none", NAMEDATALEN) != 0)
```

Comment:
Also it is a very edge case but what if there is a table whose name is "default" or "none"?